[DL-5330] [filter-form] Replace `Type bestuursorgaan` filter for `Type eredienst`

Poltergeistz opened the pull request:
# Description
DL-5330

This PR replaces current filter `Type bestuursorgaan` filter for `Type eredienst` so users can look for recognized worship types when they're already filtering on `Type bestuur` leaving them 6 options to pick from _(Anglicaanse, Islamitische, Israëlitische, Orthodox, Protestantse of Rooms-Katholiek)_.

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

# Related services

- [frontend-worship-decisions](https://github.com/lblod/frontend-worship-decisions)
- mu-cl-resource
- [toezicht-search-query-management](https://github.com/lblod/toezicht-search-query-management-service)

# How to test (From Loket app)

1. drc up -d 
2. Make sure the seach-query-management and frontend services have been pulled.
a. You can always proxy the ember app instead of pulling the image if that doesn't work
3. Log as ABB.
4. Apply type bestuur filter then type eredienst based on the 6 options.
5. It should show results.

**See Examples here**

![Screenshot 2024-06-10 at 18-04-26 Erediensten Toezichtsdatabank](https://github.com/lblod/app-worship-decisions-database/assets/38471754/3326c9c4-1098-43ff-9521-04a0c32e81bf)

![Screenshot 2024-06-10 at 18-04-47 Erediensten Toezichtsdatabank](https://github.com/lblod/app-worship-decisions-database/assets/38471754/f0aa0341-9ba3-45b5-be6c-e51443a3472c)


# What to check

- Typos 

# Links to other PR's

- https://github.com/lblod/toezicht-search-query-management-service/pull/1
- https://github.com/lblod/app-worship-decisions-database/pull/81

# Notes

- Make proper release + CHANGELOG

Poltergeistz committed:
Changing bestuursorgan filter for type eredienst

Type bestuursorgan isn't used that much by users, so it's requested to change it for type eredienst

Poltergeistz committed:
Release v0.12.0-rc.1

Poltergeistz committed:
Linting

Windvis commented on app/models/bestuurseenheid.js:
```diff
@@ -26,6 +26,12 @@ export default class Bestuurseenheid extends Model {
+  @belongsTo('recognized-worship-type', {
+    async: true,
+    inverse: null,
+    optional: true,
```
I don't think `optional` is needed? It's the first time I see it and I can't find it in the API docs :smile: . Did a warning / error message tell you to add it or something?

Windvis replied:
Also, if the relationship isn't actually used I wouldn't even add it to the model class. Less code to maintain in the future.

Poltergeistz replied:
You're right, I was just playing around and I forgot that it doesn't exists in ember, my bad :sweat_smile:

Windvis commented on app/models/recognized-worship-type.js:
```diff
@@ -0,0 +1,6 @@
+import Model, { attr } from '@ember-data/model';
+
+export default class RecognizedWorshipType extends Model {
```
If the relationship is removed, this can probably go as well?

Poltergeistz replied:
I think yes since only the route is called, I'll quickly check

Poltergeistz replied:
Works without it

Windvis commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "frontend-worship-decisions",
-  "version": "0.11.0",
+  "version": "0.12.0-rc.1",
```
I would roll this commit back, since you used a `feature/*` prefix for the branch name, [it already built an image you can use](https://build.redpencil.io/repos/1321/pipeline/107/3). I would use that instead. Tagging versions in PRs isn't a good practice IMO and should be avoided.

Poltergeistz replied:
Check, I've used that only for convenience but I'll avoid doing it.

Windvis replied:
Yea, I understand. It's just that the "feature" image builds fill the same roll and you don't have to create tags that are potentially incorrect (for example, if this PR wasn't merged for some reason).

Poltergeistz committed:
Removing recognizedWorshipType relationship and model

It's not needed here at the moment as the submissions route uses filter to fetch data

Poltergeistz committed:
Revert "Release v0.12.0-rc.1"

This reverts commit 0dbfeed41a36e77a72f93792e2c9cc1995a70726.

Windvis commented:
Haven't been able to test due to that missing image of the other service, but the code change looks good :+1: .
